aws/ci_build

```diff
@@ -67,7 +68,31 @@ def build
 
     RakeUtils.git_fetch
     count = RakeUtils.git_update_count
+
+    # Use a lock file to prevent git-pull and unison, both modify files in local
+    # repo, to run at the same time on the staging environment.
+    #
+    # Context: On the staging environment, we use the unison tool to sync content
+    # between local Dropbox and git repo. It works most of the time. However, we've
+    # seen half-dozen cases of Dropbox content being reverted. Those cases happened
+    # a few hours or even a few days after content editors had made file changes.
+    # They mostly coincided with the starts of new builds.
+    # This change is a speculative fix to prevent any potential conflicts between
+    # git-pull and unison.
+    #
+    if Rails.env.staging?
+      FileUtils.touch GIT_PULL_STARTED_PATH
+      # Wait a few seconds to give the unison tool time to finish.
+      # Unison is fast and usually finishes in less than 1 second.
+      sleep 2
+    end
+
     RakeUtils.git_pull if count > 0
```

Comment:
If this method Raises an Error, we probably won't run the logic to remove the lock file. Maybe we should put the lock file removal logic in an `ensure` block?